src/main/java/org/jenkinsci/plugins/workflow/graph/StandardGraphLookupView.java

```diff
@@ -33,9 +30,22 @@
     /** Map a node to its nearest enclosing block */
     HashMap<String, String> nearestEnclosingBlock = new HashMap<String, String>();
 
+    /** Map a block start node to its immediate children nodes, including AtomNode, BlockStartNode, and BlockEndNode */
+    HashMap<String, List<String>> blockChildren = new HashMap<>();
```

Comment:
Not a fan of caching this, since it's fairly easy and fast to do a quick iteration pass once we have the start and end -- that can even be exposed as an Iterator<FlowNode> and never materialized into a concrete list, due to the magic of FlowScanners -- they're Iterables so you can do a for :: each over the flownodes in a scanner, directly. 

Yes, that scan is O(n), but you're going to be iterating over children anyway.

Reply:
Yeah, I honestly wasn't sure what the right approach would be, but decided to bang out the prototype just in case.